### PR TITLE
Fix Clearpool & Ribbon poolFactory contracts

### PR DESCRIPTION
**Pull request by charlielewisme**
## What?
Repair Clearpool and Ribbon git history

## How?
Cherry-picked the two commits that got accidentally reverted
```
git checkout master
git pull
git checkout -b fix/clearpool-ribbon-poolFactory
git cherry-pick 0e71571
git cherry-pick 4b95810
git push -u origin fix/clearpool-ribbon-poolFactory
```

## Related PRs
https://github.com/blockchain-etl/ethereum-etl-airflow/pull/521
https://github.com/blockchain-etl/ethereum-etl-airflow/pull/522
https://github.com/blockchain-etl/ethereum-etl-airflow/pull/523
https://github.com/blockchain-etl/ethereum-etl-airflow/pull/524

### CODE REVIEW

**Review by TimNooren — APPROVED** [review]

**Comment by charlielewisme** [comment]
to: @imrankhan37 
cc: @TimNooren

TLDR; please rebase commits (in this case, squash) to keep the git history clean.

...

After merging, we ran the `ethereum_parse_clearpool_dag` and `ethereum_parse_ribbon_dag` and they failed due to `Dependency PoolFactory_event_PoolCreated not found`.

We had a git snafu basically. Let's look at just the Clearpool example (Ribbon is basically similar). You have 2 PRs, first [PR523](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/523) contains [0e71571](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/523/commits/0e71571e83073d88b04db6a52bdb123873359db3) and this gets merged. Then the second [PR524](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/524) contains [0e71571](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/524/commits/0e71571e83073d88b04db6a52bdb123873359db3) followed by [db46e16](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/524/commits/db46e16432b9489477be6b8dfb48fbd8dc36aa57). When it gets merged it does it serially, and the the first commit [0e71571](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/523/commits/0e71571e83073d88b04db6a52bdb123873359db3) is merged -- this does nothing, it's literally the same as HEAD. The second commit  [db46e16](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/524/commits/db46e16432b9489477be6b8dfb48fbd8dc36aa57) gets merged. But this commit does two things, it reverts the changes in  [0e71571](https://github.com/blockchain-etl/ethereum-etl-airflow/pull/523/commits/0e71571e83073d88b04db6a52bdb123873359db3) then adds its own files.

Now, to GitHub's "Files Changed" tab (and to Imran and Tim the reviewer), the second PR looks like it's only adding files because the two commits in the PR collectively create that diff; but that's not what happens when we merge.

Or another way to think about it is Github shows you the changes relative to the base of the feature branch, not relative to the actual target branch (or something to that effect at least). Which is super annoying since it's non-trivial to check this every time.

...

The ideal situation would have been for Imran to have rebased the 2 commits locally, squashing them into a single commit before raising the PR. Then the diff would look like it did, and the merge would add files only.

The other alternative is not to separate into 2 PRs but keep them together as one.

@imrankhan37 Make sense? Keen to avoid this in future.
